app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -65,151 +63,174 @@ public void onServiceConnected(ComponentName className, IBinder service) {
+                mPlayPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
             }
         }
-
         // Called when the connection with the service disconnects unexpectedly
         public void onServiceDisconnected(ComponentName className) {
-            Log.e(TAG, "onServiceDisconnected");
             boundToRadioService = false;
         }
     };
 
+    /**
+     *  Generate main layout, navigation drawer for the main activity
+     * @param savedInstanceState
+     */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         // set the layout to use
         setContentView(R.layout.activity_main);
-
-        setupNavigation(); // setup the nav drawer and navigation functionality
+        // setup the nav drawer and navigation functionality
+        setupNavigation();
         setupFragments(savedInstanceState);
         setupPlaybackToolbar(); // setup the playback toolbar
 
         // update the schedule if connected to internet
         if (NetworkState.isOnline(this))
             updateSchedule();
-
     }
 
+    /**
+     *Bind the radio services after the activity is being displayed to the user
+     */
     @Override
     protected void onStart() {
         super.onStart();
-        Log.d(TAG, "onStart");
-
-        // bind to the radio service
+        // Start an intent to bind to the radio service
         Intent intent = new Intent(this, RadioService.class);
         startService(intent);
         bindService(intent, mConnection, BIND_AUTO_CREATE);
-
-        if (backStack.peek() != R.id.visualizer)
+        if (mBackStack.peek() != R.id.visualizer)
             updateShowNamePlaybackToolbar();
     }
 
+    /**
+     *Set up icon for the pause and play button
+     */
     @Override
     protected void onResume() {
         super.onResume();
         // if the stream is playing, then stop the notification
         if (boundToRadioService && radioService.isPlaying()) {
-            playPauseButton.setImageResource(R.drawable.ic_pause_white_24dp);
+            mPlayPauseButton.setImageResource(R.drawable.ic_pause_white_24dp);
         } else {
-            playPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
+            mPlayPauseButton.setImageResource(R.drawable.ic_play_arrow_white_24dp);
         }
     }
 
+    /**
+     * Stop the activity
+     */
     @Override
     protected void onStop() {
         super.onStop();
         // if the stream is playing, then start the notification
-        if (boundToRadioService && radioService.isPlaying())
+        if (boundToRadioService && radioService.isPlaying()) {
             radioService.showNotification();
-
+        }
         // unbind the radio service
         unbindService(mConnection);
-
         boundToRadioService = false;
 
     }
 
+    /**
+     * Set up the playback bar on the bottom of the activity
+     */
     private void setupPlaybackToolbar() {
```

Comment:
I like that you comment this method. This method is supposed to set the Bottom navigation. let's keep it at exactly that. It's very long and we need to break it into smaller chunks. We can separate all of the onClick listeners into a `setButtomNavigationOnClickListeners()` and that'll cut it down.